src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/DownstreamFailureCause.java

```diff
@@ -37,35 +40,60 @@ public final class DownstreamFailureCause extends CauseOfInterruption {
 
     private static final long serialVersionUID = 1;
 
-    private final String id;
+    private final String jobFullName;
+    private final int buildNumber;
+    private final String url;
+    private final Result result;
+    // Only set on deserialized causes from before the above fields were added.
+    @Deprecated
+    private String id;
 
     DownstreamFailureCause(Run<?, ?> downstream) {
-        id = downstream.getExternalizableId();
+        jobFullName = downstream.getParent().getFullName();
+        buildNumber = downstream.getNumber();
+        url = downstream.getUrl();
+        result = downstream.getResult();
     }
 
     public @CheckForNull Run<?, ?> getDownstreamBuild() {
-        return Run.fromExternalizableId(id);
+        if (id != null) {
+            return Run.fromExternalizableId(id);
+        }
+        Job<?, ?> job = Jenkins.get().getItemByFullName(jobFullName, Job.class);
+        return job != null ? job.getBuildByNumber(buildNumber) : null;
     }
 
     @Override public void print(TaskListener listener) {
         String description;
-        Run<?, ?> downstream = getDownstreamBuild();
-        if (downstream != null) {
-            // encodeTo(Run) calls getDisplayName, which does not include the project name.
-            description = ModelHyperlinkNote.encodeTo("/" + downstream.getUrl(), downstream.getFullDisplayName()) + " completed with status " + downstream.getResult() + " (propagate: false to ignore)";
+        if (id != null) {
+            Run<?, ?> downstream = getDownstreamBuild();
+            if (downstream != null) {
+                // encodeTo(Run) calls getDisplayName, which does not include the project name.
+                description = description(ModelHyperlinkNote.encodeTo("/" + downstream.getUrl(), downstream.getFullDisplayName()), downstream.getResult());
+            } else {
+                description = "Downstream build was not stable (propagate: false to ignore)";
+            }
         } else {
-            description = "Downstream build was not stable (propagate: false to ignore)";
+            description = description(ModelHyperlinkNote.encodeTo("/" + url, jobFullName + " #" + buildNumber), result);
```

Comment:
Note that this is swapping `fullDisplayName` for `fullName`, so ` » ` ⇒ `/` as well as ignoring any display names set on the job and/or folders.